packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -202,7 +202,13 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
         resolve(GestureDisposition.rejected);
         stopTrackingPointer(primaryPointer);
       } else {
-        handlePrimaryPointer(event);
+        try {
+          handlePrimaryPointer(event);
+        } catch (e) {
+          resolve(GestureDisposition.rejected);
+          stopTrackingPointer(primaryPointer);
+          rethrow;
```

Comment:
The goal is that gesture recognizers should be reset to a known good state if event handler code supplied by the app throws.

Is there a better way to do this?